src/client/runner/step-iterator.js

```diff
@@ -313,103 +321,93 @@ StepIterator.prototype._checkIFrame = function (element, callback) {
-                elements,
-                function (element, asyncCallback) {
-                    //NOTE: since v.14.1.5 it's recommended to run actions with the inIFrame function. But we should support old-style iframes
-                    //using, so, it'll be resolved here.
-                    iterator._checkIFrame(element, function (iframe) {
-                        if (!iframe) {
-                            actionBarrier.waitActionSideEffectsCompletion(function (barrierCallback) {
-                                action(element, barrierCallback);
-                            }, asyncCallback);
-                        }
-                        else {
-                            var iFrameStartXhrBarrier = iframe.contentWindow[xhrBarrier.GLOBAL_START_XHR_BARRIER],
-                                iFrameWaitXhrBarrier  = iframe.contentWindow[xhrBarrier.GLOBAL_WAIT_XHR_BARRIER];
-
-                            actionBarrier.waitActionSideEffectsCompletion(function (barrierCallback) {
-                                var iFrameBeforeUnloadRaised = false;
-
-                                iterator.iFrameActionCallback = function () {
-                                    iterator.iFrameActionCallback = null;
-                                    iterator.waitedIFrame         = null;
-                                    barrierCallback();
-                                };
-
-                                iterator.waitedIFrame = iframe;
-
-                                iFrameStartXhrBarrier(function () {
-                                    if (!iFrameBeforeUnloadRaised)
-                                        iterator.iFrameActionCallback();
-                                });
-
-                                function onBeforeUnload () {
-                                    nativeMethods.windowRemoveEventListener.call(iframe.contentWindow, 'beforeunload', onBeforeUnload);
-                                    iFrameBeforeUnloadRaised = true;
-                                }
-
-                                nativeMethods.windowAddEventListener.call(iframe.contentWindow, 'beforeunload', onBeforeUnload, true);
-
-                                action(element, function () {
-                                    iFrameWaitXhrBarrier();
-                                }, iframe);
-                            }, asyncCallback);
-                        }
-                    });
-                },
-                function () {
-                    if (iterator.state.stopped)
-                        return;
+    iterator.state.inAsyncAction = true;
 
-                    callback();
+    var seriesActionsRun = function (elements, callback) {
+        async.forEachSeries(
+            elements,
+            function (element, asyncCallback) {
+                //NOTE: since v.14.1.5 it's recommended to run actions with the inIFrame function. But we should support old-style iframes
+                //using, so, it'll be resolved here.
```

Comment:
```
                //NOTE: since v.14.1.5 it's recommended to run actions using the inIFrame function. But we should support old-style iframes,
                //so it'll be resolved here.
```